pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -840,7 +840,7 @@ func (nc *Controller) monitorNodeHealth(ctx context.Context) error {
 				// Report node event only once when status changed.
 				controllerutil.RecordNodeStatusChange(nc.recorder, node, "NodeNotReady")
 				fallthrough
-			case needsRetry && observedReadyCondition.Status != v1.ConditionTrue:
+			case needsRetry && currentReadyCondition.Status != v1.ConditionTrue:
```

Comment:
Suppose there are the following cases:
1. in current iteration, currentReadyCondition.Status != v1.ConditionTrue and observedReadyCondition.Status == v1.ConditionTrue, that means Ready -> NotReady, then fallthrough to MarkPodNotReady, if mark failed, then store to nodeRetry
2. in the next iteration,  if the node becomes ready, that means observedReadyCondition and currentReadyCondition are both ready, if the node is still not ready, then currentReadyCondition could not become ready(see `tryUpdateNodeHealth`, Ready ConditionTrue can only set by kubelet itself)
so we need not do this change